spring-cloud-kubernetes-core/src/main/java/org/springframework/cloud/kubernetes/profile/KubernetesApplicationContextInitializer.java

```diff
@@ -26,30 +27,38 @@
 public class KubernetesApplicationContextInitializer implements
         ApplicationContextInitializer<ConfigurableApplicationContext>, Ordered {
 
-    private final KubernetesProfileApplicationListener listener;
     private static final int ORDER = 100;
 
-    public KubernetesApplicationContextInitializer() {
-        //If we are inside Kubernetes this should be perfectly valid.
-        //If not then we won't add the Kubernetes profile anyway.
-        this(new StandardPodUtils(new DefaultKubernetesClient()));
-    }
+    private final Supplier<KubernetesProfileApplicationListener> listenerSupplier;
 
-    public KubernetesApplicationContextInitializer(PodUtils utils) {
-        this(new KubernetesProfileApplicationListener(utils));
-    }
+	public KubernetesApplicationContextInitializer() {
+		this(LazilyInstantiate.using(() ->
+			//If we are inside Kubernetes this should be perfectly valid.
+			//If not then we won't add the Kubernetes profile anyway.
+			new KubernetesProfileApplicationListener(
+				new StandardPodUtils(new DefaultKubernetesClient()))
+		));
+	}
 
-    public KubernetesApplicationContextInitializer(KubernetesProfileApplicationListener listener) {
-        this.listener = listener;
-    }
+	public KubernetesApplicationContextInitializer(
+		Supplier<KubernetesProfileApplicationListener> listenerSupplier) {
+		this.listenerSupplier = listenerSupplier;
+	}
 
-    @Override
+	@Override
     public int getOrder() {
         return ORDER;
     }
 
     @Override
     public void initialize(ConfigurableApplicationContext applicationContext) {
-        listener.addKubernetesProfile(applicationContext.getEnvironment());
+    	if(isKubernetesEnabled(applicationContext)){
```

Comment:
Do we need that? Isn't lazy initialization enough?

Reply:
Now that you mention it, that should be enough :).
I will update the PR

Reply:
PR updated